service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -87,6 +87,7 @@ public UpdateResult<Run> updateRuns(
               Run::runId,
               r ->
                   r.status().nonTerminal()
+                      && r.engineId() != null
```

Comment:
This is fine for now. I wonder if longer term we want something like `r.status().isSubmittedToEngine()`. But fine for now 👍

Reply:
Thinking a little more... because it might be nice to precompute engine IDs and store them in the DB, even before we submit the workflow to the engine. 

That would let us be idempotent with workflow submissions across CBAS restarts, for example.

Reply:
TL;DR: if it's possible to do this with something like a `r.status().isSubmittedToEngine()` then I think that would be better. But feel free to leave this as is if that turns out to be difficult for some reason

Reply:
Talked offline and this would be included in a follow-up ticket